core/src/main/java/org/radargun/stages/cache/background/LogChecker.java

```diff
@@ -110,32 +108,36 @@ public void run() {
                   LastOperation lastCheck = (LastOperation) last;
                   record = newRecord(record, lastCheck.getOperationId(), lastCheck.getSeed());
                }
+               checkIgnoreRecord(record);
                if (record.getOperationId() != 0) {
-                  log.debugf("Check for thread %d continues from operation %d",
-                        record.getThreadId(), record.getOperationId());
-               }
-            }
-            if (ignoreDeadCheckers) {
-               Object ignored = basicCache.get(ignoredKey(slaveIndex, record.getThreadId()));
-               if (ignored != null && record.getOperationId() <= (Long) ignored) {
-                  log.debugf("Ignoring operations %d - %d for thread %d", record.getOperationId(), ignored, record.getThreadId());
-                  while (record.getOperationId() <= (Long) ignored) {
-                     record.next();
-                  }
+                  log.tracef("Check for thread %d continues from operation %d",
+                             record.getThreadId(), record.getOperationId());
                }
             }
             if (trace) {
                log.tracef("Checking operation %d for thread %d on key %d (%s)",
-                     record.getOperationId(), record.getThreadId(), record.getKeyId(), keyGenerator.generateKey(record.getKeyId()));
+                          record.getOperationId(), record.getThreadId(), record.getKeyId(), keyGenerator.generateKey(record.getKeyId()));
             }
             boolean notification = record.hasNotification(record.getOperationId());
             Object value = findValue(record);
             boolean contains = containsOperation(value, record);
+            // TODO move to PrivateLogLogic?
+            // If check is unsuccessful, verify whether complementary key (k') containing current record operation does exist (every 10 unsuccessful
```

Comment:
Operation should not be confirmed until delayed removes are executed as well, am I missing something?